.github/workflows/main.yml

```diff
@@ -0,0 +1,20 @@
+on: [pull_request]
+
+jobs:
+  lint-and-tests:
+    name: 'Lint and Test'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./app_feup
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '12.x'
+      - uses: subosito/flutter-action@v1
+        with:
+          flutter-version: '1.12.13'
+      - run: flutter pub get
+      - run: flutter test
+      - run: flutter analyze --no-pub --preamble .
```

Comment:
Now that we've changed to a flutter action, why don't we divide this job into two? It would be better to parallelize them imo